packages/java/src/browser/java-client-contribution.ts

```diff
@@ -102,9 +106,15 @@ export class JavaClientContribution extends BaseLanguageClientContribution {
         });
     }
 
+    async waitForActivation(app: FrontendApplication): Promise<void> {
+        await super.waitForActivation(app);
+        this.javaExtensions = await this.javaService.collectJavaExtensions();
+    }
+
     protected createOptions(): LanguageClientOptions {
         const options = super.createOptions();
         options.initializationOptions = {
+            bundles: this.javaExtensions,
```

Comment:
If I understand correctly, the only reason to have a front-end contribution at all is to populate this field. In my mind, it would be nicer to intercept the init message on the back end and to populate the field there. If possible, we should not let these implementation details dictate architecture.

Reply:
Yes, you understand correctly. Is there a way to intercept the init command on the server?

Reply:
https://github.com/theia-ide/theia/blob/cc6d2d3b2b409c0d92a33661c2b9272c992461f3/packages/languages/src/node/language-server-contribution.ts#L62

Reply:
Thanks @akosyakov. Moved to the server side.